### DISCUSSION TITLE
kustomize-controller built with kustomize patch bringing AGE support

### DISCUSSION THREAD

**Discussion by sylr** [fluxcd/flux2#770] [Show and tell]
Hi,

For the past 2 months I've been working on a project which aims to add support for **encryption/decryption of YAML values in place** with [AGE](https://github.com/FiloSottile/age).

The goal of this was to ultimately add the same kind of AGE support into `kustomize` to do something similar to `SOPS` but simpler.

You can find binaries and examples here: https://github.com/sylr/kustomize-age

I also built a `kustomize-controller` image that you can find here: https://github.com/users/sylr/packages/container/package/kustomize-controller

So far I have ran this image instead of the official `kustomize-controller:v0.6.2` in my corporate dev env for a few days and it worked as expected.

I'd be really glad to hear your thoughts about this.

Cheers.

**Comment by hiddeco** [comment]
Are you aware that support for AGE has been added to SOPS? https://github.com/mozilla/sops/pull/688

**Reply by sylr** [reply]
I must confess I wasn't.

But anyway the whole point of my journey was to avoid SOPS. I don't like the way it marshals/unmarshals YAML and the facts that it relies on key suffixes/prefixes.

**Reply by kingdonb** [reply]
I don't think we're likely to end-around SOPS because of all the features it provides for us.

This even in spite of the current situation with maintenance of SOPS being a bit tenuous or up in the air:

* https://github.com/mozilla/sops/discussions/927
* https://github.com/fluxcd/flux2/discussions/2270

It's a good question still!

**Reply by sylr** [reply]
My AGE implementation and SOPS are not exclusive.
